### PR TITLE
[Core][Dashboard Pubsub 3/n] Migrate pubsub usages in dashboard to GCS pubsub

### PR DESCRIPTION
**Pull request by mwtian**
## Why are these changes needed?
Add support for Ray pubsub in dashboard. https://github.com/ray-project/ray/pull/20954 is the prerequisite, and contains more complete change under src/.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/node/tests/test_node.py [hunk 1]
```diff
@@ -70,6 +70,7 @@ def test_node_info(disable_aiohttp_cache, ray_start_with_dashboard):
     @ray.remote
     class Actor:
         def getpid(self):
+            print(f"actor pid={os.getpid()}")
```
Comment (iycheng) [inline dashboard/modules/node/tests/test_node.py:73]: remove?

**Comment by iycheng** [comment]
just did the early review of this one. mostly looks good!

File: dashboard/modules/node/tests/test_node.py (hunk 1 above)
Comment (mwtian) [reply dashboard/modules/node/tests/test_node.py:73]: I wanted to keep this message, or is it too noisy?

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -325,23 +349,23 @@
       --test_env=CONDA_SHLVL
       --test_env=CONDA_PREFIX
       --test_env=CONDA_DEFAULT_ENV
-      --test_env=RAY_gcs_grpc_based_pubsub=true
+      --test_env=RAY_gcs_grpc_based_pubsub=1
```
Comment (iycheng) [inline .buildkite/pipeline.yml:352]: why change this? `=true` should work I think.

**Review by fishbone — COMMENTED** [review]
LG

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (mwtian) [reply .buildkite/pipeline.yml:352]: Yes `=true` also works but `=1` just seems cleaner to me. But it makes no big difference..

**Review by fishbone — APPROVED** [review]

**Pull request merged**